apps/src/lab2/levelEditors/panels/EditPanels.tsx

```diff
@@ -0,0 +1,280 @@
+import {Panel, PanelLayout} from '@cdo/apps/panels/types';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import moduleStyles from './edit-panels.module.scss';
+import PanelsView from '@cdo/apps/panels/PanelsView';
+import {SimpleDropdown} from '@cdo/apps/componentLibrary/dropdown';
+import {
+  BodyThreeText,
+  Heading3,
+  Heading5,
+} from '@cdo/apps/componentLibrary/typography';
+import {createUuid} from '@cdo/apps/utils';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon/FontAwesomeV6Icon';
+import classNames from 'classnames';
+import ImageInput from '@cdo/apps/lib/levelbuilder/ImageInput';
+import Button from '@cdo/apps/templates/Button';
+
+const createKey = (levelName: string) => levelName + '-' + createUuid();
+
+function sanitizePanels(panels: Panel[], levelName: string) {
+  return panels.map(panel => {
+    return {
+      ...panel,
+      key: panel.key || createKey(levelName),
+    };
+  });
+}
+
+interface EditPanelsProps {
+  initialPanels: Panel[];
+  levelName: string;
+}
+
+/**
+ * Editor for the Lab2 panels level type on the level edit page.
+ */
+const EditPanels: React.FunctionComponent<EditPanelsProps> = ({
+  initialPanels,
+  levelName,
+}) => {
+  const [panels, setPanels] = useState<Panel[]>(
+    sanitizePanels(initialPanels, levelName)
+  );
+  const [toastMessage, setToastMessage] = useState('');
+  const [toastIndex, setToastIndex] = useState(0);
+
+  const updatePanel = useCallback(
+    (panel: Panel, index: number) => {
+      setPanels([...panels.slice(0, index), panel, ...panels.slice(index + 1)]);
```

Comment:
Maybe a comment explaining what this is doing? :)

Reply:
Actually looking at this again, realized I should be using the panel `key` instead of `index`!